Show progress bar when no data in adapter.  New accent color.

gbielanski opened the pull request:
Please review

gbielanski committed:
New term type and data format. Data taken from Firebase version node.

miltomasz commented on app/src/main/java/pl/example/android/garbageapp/ui/ActivitySectorBlue.java:
```diff
@@ -17,6 +19,31 @@
 public class ActivitySectorBlue extends BaseActivitySector {
 
     private ActivitySectorBlueBinding binding;
+    private RecyclerView.AdapterDataObserver mDataObserver = new RecyclerView.AdapterDataObserver() {
```
Please declare this adapter as protected and move it to BaseActivitySector

miltomasz commented on app/src/main/java/pl/example/android/garbageapp/ui/ActivitySectorBlue.java:
```diff
@@ -17,6 +19,31 @@
+            checkEmpty();
+        }
+
+        void checkEmpty() {
```
Define checkEmpty() as abstract method in BaseActivitySector. Next, override this method in each SectorActivity and use correct bindings.

miltomasz reviewed (changes requested):
Two refactorings needed. Please see comments. However I don't think this solution is good. It will work (I tested it) but it will keep showing progress bar even when we have finished request action but there's no data.

You may try just add:

binding.progressBar.setVisibility(View.VISIBLE);

in every onCreate() in ActivitySector[Green,Blue, Yellow].

Next, add binding.progressBar.setVisibility(View.GONE);

in bindDataToUI() method in every ActivitySector[Green,Blue, Yellow]. It should work with bindings framework. And it will show progress bar only when request is ongoing.

gbielanski committed:
AdapterDataObserver moved to BaseActivitySector

gbielanski commented:
As this is MVP version, progress bar behavior is implemented in this way intentionally. I opened separated ticked in Trello to cover all cases when list can be empty.
I extracted function checkEmpty to BaseActivitySector. It was good idea and cleaned the code.
